pandas/tests/test_categorical.py

```diff
@@ -3625,6 +3625,62 @@ def test_cat_accessor_api(self):
             invalid.cat
         self.assertFalse(hasattr(invalid, 'cat'))
 
+    def test_str_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.core.strings import StringMethods
+        s = Series(list('aabb'))
+        c = s.astype('category')
```

Comment:
Im not so sure: you would need to replace this 6 3liners by a much harder to read list:

```
tests = [("method", (args), {"kwarg":"whatever"}),...]
```

I still don't think it should be necessary to build tests for all string methods, as far as I can tell, I covered all code paths (`_wrap_result` and `_wrap_result_expand`)